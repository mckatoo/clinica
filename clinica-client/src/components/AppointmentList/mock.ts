export default [
  {
    id: '1',
    photo: 'jao.jpg',
    dayOfWeek: 'terça',
    dayOfMonth: 1,
    start: '08:00',
    ending: '09:00',
    patient: 'Jão',
    attendant: 'Fernanda',
    title: 'Consulta fisioterapia'
  },
  {
    id: '2',
    photo: 'ze.jpg',
    dayOfWeek: 'quarta',
    dayOfMonth: 2,
    start: '10:00',
    ending: '11:00',
    patient: 'Zé',
    attendant: 'Fernanda',
    title: 'Seção choquinho'
  },
  {
    id: '3',
    photo: 'paulo.jpg',
    dayOfWeek: 'quarta',
    dayOfMonth: 2,
    start: '15:30',
    ending: '16:30',
    patient: 'Paulo',
    attendant: 'Maria',
    title: 'Seção choquinho'
  },
  {
    id: '4',
    photo: 'paulo.jpg',
    dayOfWeek: 'sexta',
    dayOfMonth: 4,
    start: '15:30',
    ending: '16:30',
    patient: 'Paulo',
    attendant: 'Claudia',
    title: 'Seção choquinho'
  }
]
