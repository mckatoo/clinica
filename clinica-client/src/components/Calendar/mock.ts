export default [
  {
    weekday: 'seg.',
    date: '31',
    month: '5',
    year: '2021',
    totalAppointments: 3
  },
  {
    weekday: 'ter.',
    date: '8',
    month: '6',
    year: '2021',
    totalAppointments: 2
  },
  {
    weekday: 'qui.',
    date: '10',
    month: '6',
    year: '2021',
    totalAppointments: 4
  },
  {
    weekday: 'sex.',
    date: '18',
    month: '6',
    year: '2021',
    totalAppointments: 1
  },
  {
    weekday: 'seg.',
    date: '21',
    month: '6',
    year: '2021',
    totalAppointments: 9
  },
  {
    weekday: 'qui.',
    date: '1',
    month: '7',
    year: '2021',
    totalAppointments: 10
  },
  {
    weekday: 'sex.',
    date: '2',
    month: '7',
    year: '2021',
    totalAppointments: 6
  },
  {
    weekday: 's√°b.',
    date: '3',
    month: '7',
    year: '2021',
    totalAppointments: 32
  },
]
