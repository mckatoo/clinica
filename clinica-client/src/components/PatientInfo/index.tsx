import AttendingButton from 'components/AttendingButton'
import Heading from 'components/Heading'
import TextAreaField from 'components/TextAreaField'
import TextField from 'components/TextField'

import * as S from './styles'

export type PatientInfoProps = {
  id: string
  name: string
  address: string
  phone: string
  photo: string
  cpf?: string
  rg?: string
  birth_date?: string
}

const PatientInfo = ({ id, name, address, phone, photo, cpf, rg, birth_date }: PatientInfoProps) => (
  <S.Wrapper>
    <Heading color="black" lineBottom>
      {name}
    </Heading>

    <TextField label='Nome' disabled initialValue={name} />
    <TextAreaField label='EndereÃ§o' disabled initialValue={address} />

    <S.ButtonsWrapper>
      <AttendingButton id={id} size="large" hasText />
    </S.ButtonsWrapper>
  </S.Wrapper>
)

export default PatientInfo
