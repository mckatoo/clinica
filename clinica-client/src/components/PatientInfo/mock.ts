export default {
  id: '1',
  name: 'Milton Carlos Katoo',
  address: 'Rua Dr. Miguel Couto, 81\nVila Boa Esperan√ßa\nItapira-SP',
  phone: '(19) 99906 5094',
  photo: '../img/Milton.jpg',
  cpf: '290.325.438-90',
  rg: '30.896.256-4',
  birth_date: '30/12/1976'
}
