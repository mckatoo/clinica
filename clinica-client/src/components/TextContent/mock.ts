import { TextContentProps } from '.'

const TextContentMock: TextContentProps = {
  title: 'Address',
  content: `Rua Dr. Miguel Couto, 81\nVila Boa Esperan√ßa\nItapira-SP`
}

export default TextContentMock
