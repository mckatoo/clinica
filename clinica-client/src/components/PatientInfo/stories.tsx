import { Story, Meta } from '@storybook/react/types-6-0'
// import { CartContextData } from 'hooks/use-cart'
import PatientInfo, { PatientInfoProps } from '.'
import mockPatient from './mock'

export default {
  title: 'Patient/PatientInfo',
  component: PatientInfo,
  parameters: {
    backgrounds: {
      default: 'ikatoo-dark'
    }
  },
  args: mockPatient
} as Meta

export const Default: Story<PatientInfoProps> = (args) => (
  <div style={{ maxWidth: '144rem', margin: 'auto', padding: '1.5rem' }}>
    <PatientInfo {...args} />
  </div>
)

// export const IsInCart: Story<PatientInfoProps & CartContextData> = (args) => (
//   <div style={{ maxWidth: '144rem', margin: 'auto', padding: '1.5rem' }}>
//     <PatientInfo {...args} />
//   </div>
// )

// IsInCart.args = {
//   isInCart: () => true
// }
