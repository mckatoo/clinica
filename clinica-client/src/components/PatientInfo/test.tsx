import 'session.mock'

import { render, screen } from 'utils/test-utils'

import PatientInfo from './'

const props = {
  id: '1',
  name: 'Patient Name',
  address: 'Patient Address, 32',
  phone: '(19) 98877 5566',
  cpf: '234.456.123-89',
  rg: '54.234.456-4',
  birth_date: '12/05/1998',
}

describe('<PatientInfo />', () => {
  it('should render patient informations', () => {
    const { container } = render(<PatientInfo {...props} />)

    expect(
      screen.getByRole('heading', { name: /patient name/i })
    ).toBeInTheDocument()
    expect(screen.getByText('Patient Address, 32')).toBeInTheDocument()
    expect(screen.getByText('(19) 98877 5566')).toBeInTheDocument()
    expect(screen.getByText('234.456.123-89')).toBeInTheDocument()
    expect(screen.getByText('54.234.456-4')).toBeInTheDocument()
    expect(screen.getByText('12/05/1998')).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render buttons', () => {
    render(<PatientInfo {...props} />)

    expect(
      screen.getByRole('button', { name: /add to attending/i })
    ).toBeInTheDocument()
  })
})
