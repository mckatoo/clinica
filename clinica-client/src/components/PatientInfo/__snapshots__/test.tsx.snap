// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PatientInfo /> should render patient informations 1`] = `
.c6 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background: linear-gradient( 180deg,#B3181C 0%,#EF4549 50% );
  color: #EBFFFD;
  font-family: Poppins,-apple-system,BlinkMacSystemFont,'Segoe UI',Roboto,Oxygen,Ubuntu,Cantarell,'Open Sans','Helvetica Neue',sans-serif;
  border: 0;
  cursor: pointer;
  border-radius: 0.4rem;
  padding: 0.8rem;
  -webkit-text-decoration: none;
  text-decoration: none;
  height: 5rem;
  font-size: 1.6rem;
  padding: 0.8rem 4.8rem;
}

.c6:focus {
  outline: 1px dashed;
}

.c6:hover {
  background: linear-gradient(180deg,#B3281C 0%,#EF5549 50%);
}

.c6 svg {
  width: 1.5rem;
}

.c6 svg + span {
  margin-left: 0.8rem;
}

.c7 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c1 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 2rem;
}

.c1::after {
  position: absolute;
  left: 0;
  bottom: -0.5rem;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #ED3237;
}

.c3 {
  position: absolute;
  top: 1.6rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-weight: 600;
  color: #EBFFFD;
  z-index: 10;
  background-color: #F98A05;
  font-size: 1.4rem;
  padding: 0 2.4rem;
  height: 3.6rem;
  right: -2rem;
}

.c3::before {
  content: '';
  position: absolute;
  right: 0;
  border-style: solid;
  border-left-width: 0rem;
  border-right-color: transparent;
  border-bottom-color: transparent;
  border-bottom-width: 1rem;
}

.c3::before {
  border-left-color: #955303;
  border-top-color: #955303;
}

.c3::before {
  top: 3.6rem;
  border-top-width: 1rem;
  border-right-width: 2rem;
}

.c0 {
  position: relative;
  background: #EBFFFD;
  padding: 2.4rem;
}

.c0 .c2 {
  right: -1rem;
}

.c0 .c2:before {
  border-right-width: 1rem;
}

.c4 {
  font-size: 1.4rem;
  color: #8F8F8F;
  margin-bottom: 2.4rem;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 > button {
  width: 100%;
  margin-bottom: 0.8rem;
}

@media (min-width:768px) {
  .c1 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {
  .c0 .c2 {
    right: 2.4rem;
    top: 2.4rem;
    font-size: 1.8rem;
  }

  .c0 .c2:before {
    border: none;
  }
}

@media (min-width:768px) {
  .c4 {
    max-width: 77rem;
  }
}

@media (min-width:768px) {
  .c5 {
    -webkit-flex-direction: row-reverse;
    -ms-flex-direction: row-reverse;
    flex-direction: row-reverse;
  }

  .c5 > button {
    width: initial;
    margin-bottom: 0;
  }
}

<div
  class="c0"
>
  <h2
    class="c1"
    color="black"
  >
    Patient Name
  </h2>
  <div
    class="c2 c3"
    color="secondary"
  >
    12/05/2021
  </div>
  <p
    class="c4"
  >
    Patient Address, 32
  </p>
  <p
    class="c4"
  >
    (19) 98877 5566
  </p>
  <p
    class="c4"
  >
    234.456.123-89
  </p>
  <p
    class="c4"
  >
    54.234.456-4
  </p>
  <p
    class="c4"
  >
    12/05/1998
  </p>
  <div
    class="c5"
  >
    <button
      aria-label="Add to attending"
      class="c6"
    >
      <svg
        aria-hidden="true"
        class="c7"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5 22h14c1.103 0 2-.897 2-2V5c0-1.103-.897-2-2-2h-2a1 1 0 00-1-1H8a1 1 0 00-1 1H5c-1.103 0-2 .897-2 2v15c0 1.103.897 2 2 2zM5 5h2v2h10V5h2v15H5V5z"
        />
        <path
          d="M11 13.586l-1.793-1.793-1.414 1.414L11 16.414l5.207-5.207-1.414-1.414z"
        />
      </svg>
      <span>
        Add to attending
      </span>
    </button>
  </div>
</div>
`;
